packages/commonwealth/client/scripts/views/components/sidebar/CommunitySection/CommunitySection.tsx

```diff
@@ -53,6 +57,12 @@ export const CommunitySection = ({ showSkeleton }: CommunitySectionProps) => {
   const topicIdsIncludedInContest =
     getUniqueTopicIdsIncludedInActiveContest(contestsData);
 
+  const communityAlerts:
+    | ReadonlyArray<z.infer<typeof CommunityAlert>>
+    | undefined = useCommunityAlertsQuery({
+    enabled: user.isLoggedIn && !!app.chain,
+  }).data as unknown as ReadonlyArray<z.infer<typeof CommunityAlert>>;
+
```

Comment:
looks very complicated. This is what we do to use a hook

```
  const { data: communityAlerts } = useCommunityAlertsQuery({
    enabled: user.isLoggedIn && !!app.chain,
  });
```
The reason we use trpc is to not type explicitly the response, it just does not make sense.
In theory, in SubscriptionButton, communityAlerts prop should have the same type as communityAlerts data from the trpc

Reply:
The problem is tRPC doesn't play well with dates. If I don't use type coercion here I get this type error:
```
Types of property created_at are incompatible.
Type string | undefined is not assignable to type Date | undefined
```

I think @burtonator is much more familiar with this problem as he first discovered and attempted to resolve it. If you have a solution @masvelio I'm happy to implement it here but in the meantime I can't use the pattern you describe because it doesn't allow me to coerce the type f `communityAlerts`.

Reply:
https://github.com/hicommonwealth/commonwealth/issues/7866 Please resolve this if you agree we can handle this issue separately @masvelio.

Reply:
I see, but for me we should resolve the issue ASAP. Otherwise we are resolving one problem and creating another one + the readability is terrible. @burtonator would you be able to work on #7866?

Reply:
Yeah. I'll take a look again and see if we can fix it... I might be able to come up with a workaround.